src/mscorlib/src/System/Threading/ThreadPool.cs

```diff
@@ -441,6 +447,11 @@ public void Enqueue(IThreadPoolWorkItem callback, bool forceGlobal)
             if (null != tl)
             {
                 tl.workStealingQueue.LocalPush(callback);
+
+                if (Volatile.Read(ref WorkStealingQueueList.wsqActive) != WorkStealingQueueList.WsqNowActive)
```

Comment:
Yea I think that is safe. If you don't mind, could you please add some comments about the reasoning behind these choices? I'm sure we will forget quite soon.